[CI] DateUtilsTests testTimezoneIds failing

**Build Scans:**
- [elasticsearch-periodic #6937 / zulu8_java-matrix](https://gradle-enterprise.elastic.co/s/t4zoudjodpxjg)
- [elasticsearch-periodic #6937 / corretto11_java-matrix](https://gradle-enterprise.elastic.co/s/bs4nhkou3thjk)
- [elasticsearch-periodic #6937 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/ceskzthps2uyq)
- [elasticsearch-periodic #6930 / adoptopenjdk11_java-matrix](https://gradle-enterprise.elastic.co/s/xcozqn5ompluq)
- [elasticsearch-periodic #6930 / corretto11_java-matrix](https://gradle-enterprise.elastic.co/s/wu2bs74tqqg7u)
- [elasticsearch-periodic #6930 / zulu8_java-matrix](https://gradle-enterprise.elastic.co/s/kwq4ndhmvshay)
- [elasticsearch-periodic #6930 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/zeirjnetlifoi)
- [elasticsearch-periodic #6930 / zulu11_java-matrix](https://gradle-enterprise.elastic.co/s/wzd6pylloklk4)
- [elasticsearch-periodic #6923 / adoptopenjdk11_java-matrix](https://gradle-enterprise.elastic.co/s/a5dcfs3httljq)
- [elasticsearch-periodic #6923 / zulu11_java-matrix](https://gradle-enterprise.elastic.co/s/gcvryh2sm5xfc)

**Reproduction Line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.common.time.DateUtilsTests.testTimezoneIds" -Dtests.seed=981DA435DCB65EE -Dtests.locale=es-ES -Dtests.timezone=Pacific/Honolulu -Druntime.java=8
```

**Applicable branches:**
7.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.common.time.DateUtilsTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testTimezoneIds),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: WET
Expected: <0>
     but: was <3600000>
```

**Issue Reasons:**
- [7.17] 3 consecutive failures in test testTimezoneIds
- [7.17] 22 consecutive failures in step zulu8_java-matrix
- [7.17] 22 consecutive failures in step corretto11_java-matrix
- [7.17] 22 consecutive failures in step openjdk23_java-matrix
- [7.17] 21 consecutive failures in step adoptopenjdk11_java-matrix
- [7.17] 21 consecutive failures in step zulu11_java-matrix
- [7.17] 108 failures in test testTimezoneIds (11.3% fail rate in 952 executions)
- [7.17] 22 failures in step zulu8_java-matrix (100.0% fail rate in 22 executions)
- [7.17] 22 failures in step corretto11_java-matrix (100.0% fail rate in 22 executions)
- [7.17] 22 failures in step openjdk23_java-matrix (100.0% fail rate in 22 executions)
- [7.17] 21 failures in step adoptopenjdk11_java-matrix (100.0% fail rate in 21 executions)
- [7.17] 21 failures in step zulu11_java-matrix (100.0% fail rate in 21 executions)
- [7.17] 22 failures in pipeline elasticsearch-periodic (100.0% fail rate in 22 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
@@ -204,10 +204,20 @@ public abstract class ESTestCase extends LuceneTestCase {
                 "Eire",
                 "Europe/Dublin", // dublin timezone in joda does not account for DST
                 "Asia/Qostanay", // part of tzdata2018h
-                "America/Godthab", // part of tzdata2020a (maps to America/Nuuk)
-                "America/Nuuk", // part of tzdata2020a
+                "America/Bahia_Banderas",   // tzdata2024b
+                "America/Cancun",   // tzdata2024b
+                "America/Chihuahua",   // tzdata2024b
                 "America/Ciudad_Juarez", // part of tzdata2022g
+                "America/Godthab", // part of tzdata2020a (maps to America/Nuuk)
+                "America/Hermosillo",   // tzdata2024b
+                "America/Mazatlan",   // tzdata2024b
+                "America/Merida",   // tzdata2024b
+                "America/Mexico_City",   // tzdata2024b
+                "America/Monterrey",   // tzdata2024b
+                "America/Nuuk", // part of tzdata2020a
                 "America/Pangnirtung", // part of tzdata2022g
+                "America/Ojinaga",   // tzdata2024b
+                "America/Tijuana",   // tzdata2024b
                 "Europe/Kyiv", // part of tzdata2022c,
                 "Pacific/Kanton", // part of tzdata2021b
                 "Pacific/Niue",

